From f3f91faa503bc2ad66a0a72f78f1c4eb4e180077 Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@nvidia.com>
Date: Sun, 18 Jul 2021 20:36:48 +0000
Subject: [PATCH backport v4.19 102/104] mlxsw: minimal: Obtain maximum number
 of slots supported by system

Obtain maximum slot number by query MGPIR register with 'slot_index'
zero. This info is to be used to pre-allocate memory for maximum line
card number during initialization.

Signed-off-by: Vadim Pasternak <vadimp@nvidia.com>
---
 drivers/net/ethernet/mellanox/mlxsw/minimal.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/mellanox/mlxsw/minimal.c b/drivers/net/ethernet/mellanox/mlxsw/minimal.c
index 99bb8eb792d3..eab2a216c00c 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/minimal.c
+++ b/drivers/net/ethernet/mellanox/mlxsw/minimal.c
@@ -249,6 +249,11 @@ static int mlxsw_m_ports_create(struct mlxsw_m_area *mlxsw_m_area, u8 slot_index
 	if (err)
 		return err;
 
+	/* Get maximum number of slots supported by system. */
+	if (!slot_index)
+		mlxsw_m->max_modules_per_slot =
+			mlxsw_reg_mgpir_max_modules_per_slot_get(mgpir_pl);
+
 	mlxsw_reg_mgpir_unpack(mgpir_pl, NULL, NULL, NULL,
 			       &mlxsw_m_area->max_ports, NULL);
 	if (!mlxsw_m_area->max_ports)
-- 
2.20.1

